apps/resource_custom_app_integration.go

```diff
@@ -0,0 +1,77 @@
+package apps
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/oauth2"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type CustomAppIntegration struct {
+	oauth2.GetCustomAppIntegrationOutput
+	// OAuth client-secret generated by the Databricks. If this is a
+	// confidential OAuth app client-secret will be generated.
+	ClientSecret string `json:"client_secret,omitempty"`
+}
+
+func ResourceCustomAppIntegration() common.Resource {
+	s := common.StructToSchema(CustomAppIntegration{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		for _, p := range []string{"client_id", "create_time", "created_by", "creator_username", "integration_id"} {
+			common.CustomizeSchemaPath(m, p).SetComputed()
+		}
+		for _, p := range []string{"name", "scopes"} {
+			common.CustomizeSchemaPath(m, p).SetForceNew()
```

Comment:
Can we update `confidential`? If yes, then we may need to use force send fields in the update request if we can change from true to false. If no, then we need to mark it as force new